core/core-keybindings.el

```diff
@@ -46,50 +50,40 @@ used as the prefix command."
+      (when (and is-major-mode-prefix dotspacemacs-major-mode-emacs-leader-key)
+        (which-key-declare-prefixes-for-mode mode major-mode-prefix-emacs prefix-name)))))
+
+(defun spacemacs/set-keys (key def &rest bindings)
+  (while key
+    (define-key spacemacs-default-map (kbd key) def)
+    (setq key (pop bindings) def (pop bindings))))
+
+(defun spacemacs//init-major-mode-map (mode map)
+  (let ((prefix (intern (format "%s-prefix" map))))
+    (or (boundp prefix)
+        (progn
+          (eval
+           `(bind-map ,map
+              :prefix-cmd ,prefix
+              :major-modes (,mode)
+              :keys (,dotspacemacs-major-mode-emacs-leader-key
+                     ,(concat dotspacemacs-emacs-leader-key " m"))
+              :evil-keys (,dotspacemacs-major-mode-leader-key
+                          ,(concat dotspacemacs-leader-key " m"))))
+          (boundp prefix)))))
+
+(defun spacemacs/set-keys-for-major-mode (mode key def &rest bindings)
+  (let* ((map (intern (format "spacemacs-%s-map" mode))))
+    (when (spacemacs//init-major-mode-map mode map)
+      (while key
+        (define-key (symbol-value map) (kbd key) def)
+        (setq key (pop bindings) def (pop bindings))))))
 
-(define-minor-mode spacemacs-additional-leader-mode ()
-  "This mode follows the design of `evil-leader-mode' and
-complements it by added additional leader keys."
-  :init-value nil
-  :keymap nil
-  (let* ((mode-map (cdr (assoc major-mode evil-leader--mode-maps)))
-         (root-map (when spacemacs-additional-leader-mode
-                     (or mode-map evil-leader--default-map)))
-         (major-mode-map (when (and spacemacs-additional-leader-mode
-                                    mode-map)
-                           (lookup-key mode-map (kbd "m"))))
-         (state-maps '(evil-normal-state-local-map
-                       evil-motion-state-local-map
-                       evil-visual-state-local-map))
-         (emacs-state-maps '(evil-emacs-state-local-map
-                             evil-insert-state-local-map
-                             evil-normal-state-local-map
-                             evil-motion-state-local-map
-                             evil-visual-state-local-map
-                             evil-hybrid-state-local-map)))
-    (dolist (state-map state-maps)
-      (when state-map
-        (setq state-map (eval state-map))
-        (when dotspacemacs-major-mode-leader-key
-          (define-key state-map
-            (kbd dotspacemacs-major-mode-leader-key) major-mode-map))))
-    (dolist (state-map emacs-state-maps)
-      (when state-map
-        (setq state-map (eval state-map))
-        (when dotspacemacs-major-mode-emacs-leader-key
-          (define-key state-map
-            (kbd dotspacemacs-major-mode-emacs-leader-key) major-mode-map))
-        (define-key state-map (kbd dotspacemacs-emacs-leader-key) root-map)))))
+(defalias 'spacemacs/set-keys-for-mode 'spacemacs/set-keys-for-major-mode)
```

Comment:
Seems like `spacemacs/set-keys-for-major-mode` is used only once, consider defined directly the function with name `spacemacs/set-keys-for-mode` (renamed accordingly if we change it, see above comments).